dask/array/tests/test_array_core.py

```diff
@@ -3615,12 +3615,298 @@ def test_setitem_2d():
     assert_eq(x, dx)
 
 
+def test_setitem_extended_API():
+    # 0-d array
+    x = np.array(9)
+    dx = da.from_array(9)
+
+    x[()] = -1
+    dx[()] = -1
+    assert_eq(x, dx.compute())
+
+    x[...] = -11
+    dx[...] = -11
+    assert_eq(x, dx.compute())
+
+    # 1-d array
+    x = np.arange(10)
+    dx = da.from_array(x.copy(), chunks=(4, 6))
+
+    x[2:8:2] = -1
+    dx[2:8:2] = -1
+    assert_eq(x, dx.compute())
+
+    x[...] = -11
+    dx[...] = -11
+    assert_eq(x, dx.compute())
+
+    # 2-d array
+    x = np.ma.arange(60).reshape((6, 10))
+    dx = da.from_array(x.copy(), chunks=(2, 3))
+
+    check_each_op = False
```

Comment:
You're quite right  - debuggy! Each operation is tested in any case, but if check_each_op=True then an interim compute is alos done as each "setitem" layer of the graph is added, which ~doubles the run time. If the test fails when check_each_op=False then it is impossible to know which operation caused the failure.

I could redesign the test so that the dask array is reinitialised for each __setitem__ operation:
```python
    dx = da.from_array(x.copy(), chunks=(2, 3))
    x[2:4, x[0] > 3] = -5
    dx[2:4, dx[0] > 3] = -5
    assert_eq(x, dx.compute())

    dx = da.from_array(x.copy(), chunks=(2, 3))
    x[2, x[0] < -2] = -7
    dx[2, dx[0] < -2] = -7
    assert_eq(x, dx.compute())
```
This would not build up and execute huge graph, but I think should still exercise the functionality.

Would that be OK?

Reply:
Sounds good. Maybe you can try splitting up the test in smaller functions and try `@pytest.mark.parametrize`?

Reply:
I like that approach, thanks. How about: https://github.com/dask/dask/pull/7393/commits/95fb2427f9d7d25d48bc30aa1c5a11c211955f7b